flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/ScanContext.java

```diff
@@ -75,7 +85,9 @@
   private final boolean caseSensitive;
   private final boolean exposeLocality;
   private final Long snapshotId;
+  private final StreamingStartingStrategy startingStrategy;
   private final Long startSnapshotId;
+  private final Long startSnapshotTimestamp;
```

Comment:
Should we add the other Long parameter : `endSnapshotTimestamp`?  just like `startSnapshotId` and `endSnapshotId`.

Reply:
@YesOrNo828 I thought about the `endSnapshotTimestamp`. This is only useful if we want to set an end time in the future. If it is an end time in the past, we don't really need the streaming read with a stop/end timestamp. We can just use a regular filter with batch read mode.

We can add the `endSnapshotTimestamp` feature when there is real need for it. I just didn't include it in the current PR.